superset/db_engine_specs/bigquery.py

```diff
@@ -72,11 +72,13 @@ class BigQueryEngineSpec(BaseEngineSpec):
     }
 
     @classmethod
-    def convert_dttm(cls, target_type: str, dttm: datetime) -> str:
+    def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
         tt = target_type.upper()
         if tt == "DATE":
-            return "'{}'".format(dttm.strftime("%Y-%m-%d"))
-        return "'{}'".format(dttm.strftime("%Y-%m-%d %H:%M:%S"))
+            return f"CAST('{dttm.date().isoformat()}' AS DATE)"
+        if tt == "TIMESTAMP":
+            return f"""CAST('{dttm.isoformat(sep=" ", timespec="seconds")}' AS TIMESTAMP)"""  # pylint: disable=line-too-long
```

Comment:
This doesn't need to be truncated to seconds or have " " separator. Also we should add similar logic for `DATETIME`, which should conform to `f"""CAST('{dttm.isoformat()}' AS DATETIME)"""`. See below:

![image](https://user-images.githubusercontent.com/33317356/67800874-c04cbd00-fa90-11e9-8880-df61e9309ac1.png)

Reply:
Thanks. Documentation for some of these conversions is challenging to find.